src/main/java/teammates/ui/controller/StudentFeedbackResultsPageData.java

```diff
@@ -1,13 +1,225 @@
+     * @return Feedback results responses table for a question and a recipient
+     */
+    private FeedbackResultsResponseTable createResponseTable(FeedbackQuestionAttributes question, 
+                                    List<FeedbackResponseAttributes> responsesBundleForRecipient) {
+        
+        List<FeedbackResultsResponse> responses = new ArrayList<FeedbackResultsResponse>();
+        String recipientName = responsesBundleForRecipient == null || responsesBundleForRecipient.isEmpty() ?
+                                 "" : bundle.getRecipientNameForResponse(question, responsesBundleForRecipient.get(0));
+     
+        for (FeedbackResponseAttributes singleResponse : responsesBundleForRecipient) {
+            String giverName = bundle.getGiverNameForResponse(question, singleResponse);
+            
+            /* Change display name to 'You' or 'Your team' if necessary */
+            if (question.giverType == FeedbackParticipantType.TEAMS) {
+                if (student.team.equals(giverName)) {
+                    giverName = "Your Team (" + giverName + ")";
+                }
+            } else if (student.email.equals(singleResponse.giverEmail)) {
+                giverName = "You";
+            }
+            
+            if (question.recipientType == FeedbackParticipantType.TEAMS) {
+                if (student.team.equals(singleResponse.recipientEmail) && 
+                      !(recipientName.startsWith("Your Team (") && recipientName.endsWith(")"))) { // To avoid duplicate replacement
+                    recipientName = "Your Team (" + recipientName + ")";
+                }
+            } else if (student.email.equals(singleResponse.recipientEmail)
+                       && student.name.equals(recipientName)) {
+                recipientName = "You";
+            }
+
+            /* If the giver is the same user, show the real name of the receiver */
+            if (giverName.equals("You") && (!recipientName.equals("You"))) {
+                recipientName = bundle.getNameForEmail(singleResponse.recipientEmail);
+            }
+            
+            String answer = singleResponse.getResponseDetails().getAnswerHtml(question.getQuestionDetails());
+            List<StudentFeedbackResultsResponseComment> comments = createStudentFeedbackResultsResponseComments(singleResponse);
+            
+            responses.add(new FeedbackResultsResponse(giverName, answer, comments));
+        }
+        return new FeedbackResultsResponseTable(recipientName, responses);
+    }
+
+    /**
+     * Creates a list of comments for a feedback results response
+     * @param singleResponse  Response for which comments are created
+     * @return Comments for the response
+     */
+    private List<StudentFeedbackResultsResponseComment> createStudentFeedbackResultsResponseComments(
+                                                            FeedbackResponseAttributes singleResponse) {
```

Comment:
Maybe just pass a `feedbackResponseId` `String` variable?
I suppose it doesn't matter too much, but the method is using only the id, after all.